pkgs/applications/misc/octoprint/default.nix

```diff
@@ -16,58 +16,60 @@ let
 
-      # Octoprint holds back jinja2 to 2.8.1 due to breaking changes.
-      # This old version does not have updated test config for pytest 4,
-      # and pypi tarball doesn't contain tests dir anyways.
+      # Built-in dependency
       (self: super: {
-        jinja2 = super.jinja2.overridePythonAttrs (oldAttrs: rec {
-          version = "2.8.1";
-          src = oldAttrs.src.override {
-            inherit version;
-            sha256 = "14aqmhkc9rw5w0v311jhixdm6ym8vsm29dhyxyrjfqxljwx1yd1m";
+        octoprint-filecheck = self.buildPythonPackage rec {
+          pname = "OctoPrint-FileCheck";
+          version = "2020.08.07";
+
+          src = fetchFromGitHub {
+            owner  = "OctoPrint";
+            repo   = "OctoPrint-FileCheck";
+            rev    = version;
+            sha256 = "05ys05l5x7d2bkg3yqrga6m65v3g5fcnnzbfab7j9w2pzjdapx5b";
           };
           doCheck = false;
-        });
+        };
+      })
 
-        httpretty = super.httpretty.overridePythonAttrs (oldAttrs: rec {
-          doCheck = false;
-        });
+      # Built-in dependency
+      (self: super: {
+        octoprint-firmwarecheck = self.buildPythonPackage rec {
+          pname = "OctoPrint-FirmwareCheck";
+          version = "2020.06.22";
 
-        celery = super.celery.overridePythonAttrs (oldAttrs: rec {
+          src = fetchFromGitHub {
+            owner  = "OctoPrint";
+            repo   = "OctoPrint-FirmwareCheck";
+            rev    = version;
+            sha256 = "19y7hrgg9z8hl7cwqkvg8nc8bk0wwrsfvjd1wawy33wn60psqv1h";
+          };
           doCheck = false;
-        });
+        };
       })
+
       (self: super: {
         octoprint = self.buildPythonPackage rec {
           pname = "OctoPrint";
-          version = "1.4.0";
+          version = "1.4.2";
 
           src = fetchFromGitHub {
-            owner  = "foosel";
+            owner  = "OctoPrint";
             repo   = "OctoPrint";
             rev    = version;
-            sha256 = "0387228544v28d69dcdg2zr5gp6qavkfr6dydpjgj5awxv3w25d5";
+            sha256 = "1bblrjwkccy1ifw7lf55g3k9lq1sqzwd49vj8bfzj2w07a7qda62";
           };
 
           propagatedBuildInputs = with super; [
-            awesome-slugify flask flask_assets rsa requests pkginfo watchdog
-            semantic-version werkzeug flaskbabel tornado
-            psutil pyserial flask_login netaddr markdown
-            pylru pyyaml sarge feedparser netifaces click websocket_client
-            scandir chainmap future wrapt monotonic emoji jinja2
-            frozendict cachelib sentry-sdk filetype markupsafe
+            octoprint-firmwarecheck octoprint-filecheck markupsafe tornado markdown rsa regex flask jinja2 flask_login flask-babel flask_assets werkzeug itsdangerous cachelib pyyaml pyserial netaddr watchdog sarge netifaces pylru pkginfo requests semantic-version psutil click feedparser future websocket_client wrapt emoji frozendict sentry-sdk filetype unidecode blinker
```

Comment:
Please keep the dependencies split over multiple lines.  Otherwise the line becomes too long.  It would also be OK to have one dependency per line.

Reply:
> Please keep the dependencies split over multiple lines. Otherwise the line becomes too long. It would also be OK to have one dependency per line.

I hope using [nixpkgs-fmt](https://github.com/nix-community/nixpkgs-fmt) is acceptable.

Reply:
Yes, of course.  Apparently nixpkgs-fmt 1.0.0 seems to format the file differently, but I'm happy with the current version.